mikrotik/internal/utils/struct_copy.go

```diff
@@ -40,13 +43,24 @@ func copyStruct(src, dest interface{}) error {
 	for i := 0; i < reflectedSrc.NumField(); i++ {
 		srcField := reflectedSrc.Field(i)
 		srcFieldType := reflectedSrc.Type().Field(i)
-		destField := reflectedDest.FieldByName(srcFieldType.Name)
 
-		_, ok := reflectedDest.Type().FieldByName(srcFieldType.Name)
-		if !ok {
+		destField := reflectedDest.FieldByNameFunc(
+			func(s string) bool {
+				return strings.EqualFold(srcFieldType.Name, s)
+			})
+		destFieldType, found := reflectedDest.Type().FieldByNameFunc(
+			func(s string) bool {
+				return strings.EqualFold(srcFieldType.Name, s)
+			})
+
+		if !destField.IsValid() || !found {
 			// skip if dest struct does not have it (by name)
 			continue
 		}
+		if srcFieldType.PkgPath != "" || destFieldType.PkgPath != "" {
+			// skip unexported fields
+			continue
+		}
```

Comment:
Would it have been helpful to have a debug log that would print out the struct's dest and src field matching (1 log at the beginning of the loop and one after the `continue` statements)? Since we are introducing behavior that could be unpredictable, I'm just thinking about ways to easily surface an error to help aid in debugging any future complications.

Reply:
done